File: scala-advent-of-code/src/test/scala/advent/twenty_fifteen/Day01NotQuiteLispTest.scala
package advent.twenty_fifteen

import org.scalatest.{FlatSpec, Matchers}

class Day01NotQuiteLispTest extends FlatSpec with Matchers {
  "2015 Day 1 - Input" should "calculate answers" in {
    val dp = new Day01NotQuiteLisp("/twenty_fifteen/Day01-NotQuiteLisp-input.txt")

    dp.part1Answer should be(74)
    dp.part2Answer should be(1795)
  }

  "2015 Day 1 - Example #1" should "calculate answers" in {
    val dp = new Day01NotQuiteLisp("/twenty_fifteen/Day01-NotQuiteLisp-example#1.txt")

    dp.part1Answer should be(0)
    dp.part2Answer should be(-1)
  }

  "2015 Day 1 - Example #2" should "calculate answers" in {
    val dp = new Day01NotQuiteLisp("/twenty_fifteen/Day01-NotQuiteLisp-example#2.txt")

    dp.part1Answer should be(0)
    dp.part2Answer should be(-1)
  }

  "2015 Day 1 - Example #3" should "calculate answers" in {
    val dp = new Day01NotQuiteLisp("/twenty_fifteen/Day01-NotQuiteLisp-example#3.txt")

    dp.part1Answer should be(3)
    dp.part2Answer should be(-1)
  }

  "2015 Day 1 - Example #4" should "calculate answers" in {
    val dp = new Day01NotQuiteLisp("/twenty_fifteen/Day01-NotQuiteLisp-example#4.txt")

    dp.part1Answer should be(3)
    dp.part2Answer should be(-1)
  }

  "2015 Day 1 - Example #5" should "calculate answers" in {
    val dp = new Day01NotQuiteLisp("/twenty_fifteen/Day01-NotQuiteLisp-example#5.txt")

    dp.part1Answer should be(3)
    dp.part2Answer should be(1)
  }

  "2015 Day 1 - Example #6" should "calculate answers" in {
    val dp = new Day01NotQuiteLisp("/twenty_fifteen/Day01-NotQuiteLisp-example#6.txt")

    dp.part1Answer should be(-1)
    dp.part2Answer should be(3)
  }

  "2015 Day 1 - Example #7" should "calculate answers" in {
    val dp = new Day01NotQuiteLisp("/twenty_fifteen/Day01-NotQuiteLisp-example#7.txt")

    dp.part1Answer should be(-1)
    dp.part2Answer should be(1)
  }

  "2015 Day 1 - Example #8" should "calculate answers" in {
    val dp = new Day01NotQuiteLisp("/twenty_fifteen/Day01-NotQuiteLisp-example#8.txt")

    dp.part1Answer should be(-3)
    dp.part2Answer should be(1)
  }

  "2015 Day 1 - Example #9" should "calculate answers" in {
    val dp = new Day01NotQuiteLisp("/twenty_fifteen/Day01-NotQuiteLisp-example#9.txt")

    dp.part1Answer should be(-3)
    dp.part2Answer should be(1)
  }
}

File: python-advent-of-code/tests/twenty_fifteen/day01_not_quite_lisp_test.py
import unittest

from advent.twenty_fifteen.day01_not_quite_list import Day01NotQuiteLisp


class Day01NotQuiteLispTestCase(unittest.TestCase):
    def test_real(self):
        dp = Day01NotQuiteLisp("data/twenty_fifteen/Day01-NotQuiteLisp-input.txt")
        self.assertEqual(74, dp.part_1_answer)
        self.assertEqual(1795, dp.part_2_answer)

    def test_example_1(self):
        dp = Day01NotQuiteLisp("data/twenty_fifteen/Day01-NotQuiteLisp-example#1.txt")
        self.assertEqual(0, dp.part_1_answer)
        self.assertEqual(-1, dp.part_2_answer)

    def test_example_2(self):
        dp = Day01NotQuiteLisp("data/twenty_fifteen/Day01-NotQuiteLisp-example#2.txt")
        self.assertEqual(0, dp.part_1_answer)
        self.assertEqual(-1, dp.part_2_answer)

    def test_example_3(self):
        dp = Day01NotQuiteLisp("data/twenty_fifteen/Day01-NotQuiteLisp-example#3.txt")
        self.assertEqual(3, dp.part_1_answer)
        self.assertEqual(-1, dp.part_2_answer)

    def test_example_4(self):
        dp = Day01NotQuiteLisp("data/twenty_fifteen/Day01-NotQuiteLisp-example#4.txt")
        self.assertEqual(3, dp.part_1_answer)
        self.assertEqual(-1, dp.part_2_answer)

    def test_example_5(self):
        dp = Day01NotQuiteLisp("data/twenty_fifteen/Day01-NotQuiteLisp-example#5.txt")
        self.assertEqual(3, dp.part_1_answer)
        self.assertEqual(1, dp.part_2_answer)

    def test_example_6(self):
        dp = Day01NotQuiteLisp("data/twenty_fifteen/Day01-NotQuiteLisp-example#6.txt")
        self.assertEqual(-1, dp.part_1_answer)
        self.assertEqual(3, dp.part_2_answer)

    def test_example_7(self):
        dp = Day01NotQuiteLisp("data/twenty_fifteen/Day01-NotQuiteLisp-example#7.txt")
        self.assertEqual(-1, dp.part_1_answer)
        self.assertEqual(1, dp.part_2_answer)

    def test_example_8(self):
        dp = Day01NotQuiteLisp("data/twenty_fifteen/Day01-NotQuiteLisp-example#8.txt")
        self.assertEqual(-3, dp.part_1_answer)
        self.assertEqual(1, dp.part_2_answer)

    def test_example_9(self):
        dp = Day01NotQuiteLisp("data/twenty_fifteen/Day01-NotQuiteLisp-example#9.txt")
        self.assertEqual(-3, dp.part_1_answer)
        self.assertEqual(1, dp.part_2_answer)


if __name__ == '__main__':
    unittest.main()
